driver/ipam_driver.go

```diff
@@ -179,9 +179,11 @@ func (i IpamDriver) RequestAddress(request *ipam.RequestAddressRequest) (*ipam.R
 			version = ipNet.Version()
 			if version == 4 {
 				poolV4 = []caliconet.IPNet{caliconet.IPNet{IPNet: pool.Metadata.CIDR.IPNet}}
+				numIPv4 = 1
```

Comment:
`AutoAssign()` does not check `IPv4Pools`/`IPv6Pools`
https://github.com/projectcalico/libcalico-go/blob/release-v2.6/lib/client/ipam.go#L116